lib/address.py

```diff
@@ -240,15 +240,33 @@ def from_string(self, string):
                 script.extend(Script.push_data(binascii.unhexlify(word)))
         return ScriptOutput(bytes(script))
 
-    def to_ui_string(self):
-        '''Convert to a hexadecimal string.'''
+    def to_ui_string(self, hex_only = False):
+        '''Convert to user-readable OP-codes (plus text), or to a hexadecimal string if that fails.
+           Note that this function is the inverse of from_string() only if called with hex_only = True!'''
+        if self.script and not hex_only:
+            try:
+                ret = ''
+                ops = Script.get_ops(self.script)
+                for op in ops:
+                    if ret: ret += ", "
+                    def lookup(x):
+                        return OpCodes.reverseLookup.get(x, ('('+str(x)+')'))
+                    if isinstance(op, tuple):
+                        ret += lookup(op[0]) + " " + op[1].decode('utf-8')
+                    elif isinstance(op, int):
+                        ret += lookup(op)
+                    else:
+                        ret += '[' + str(op) + ']'
+                return ret
+            except:
```

Comment:
I'm not a big fan of bare except clauses.  If someone modifies the code to have an error you didn't expect, or one happens anyway, it's going to fail silently.  I'd rather you caught the exceptions you expect to happen, and suppress only those.

Reply:
Yeah you're right.  I'll change it.